jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/SQLServerQueryRewriter.java

```diff
@@ -116,6 +116,14 @@ public String rewriteFilterItem(FilterItem item) {
                 final String dateTimeValue = "CAST('" + format.format(date) + "' AS DATETIME)";
 
                 sb.append(dateTimeValue);
+
+                //Remove TIMESTAMP token as SQL Server doesn't support it
+                int timestampIndex = sb.lastIndexOf("TIMESTAMP");
```

Comment:
Hmmm, weird. I guess it could happen in other cases, then. My testing also seemed to show that the SqlServerQueryRewriter more or less deferred to the DefaultQueryRewriter directly after looking at the types etc..

@olafgarten I'm afraid you'll have to create a unit test showing the problem before we can act on it.